metals/src/main/scala/scala/meta/internal/metals/ScalafixProvider.scala

```diff
@@ -22,65 +25,98 @@ case class ScalafixProvider(
     workspace: AbsolutePath,
     embedded: Embedded,
     statusBar: StatusBar,
+    compilations: Compilations,
     icons: Icons,
-    languageClient: MetalsLanguageClient
-) {
+    languageClient: MetalsLanguageClient,
+    buildTargets: BuildTargets
+)(implicit ec: ExecutionContext) {
   import ScalafixProvider._
   private val scalafixCache = TrieMap.empty[ScalaBinaryVersion, Scalafix]
   private val organizeImportRuleCache =
-    TrieMap.empty[ScalaBinaryVersion, List[Path]]
+    TrieMap.empty[ScalaBinaryVersion, URLClassLoader]
```

Comment:
Changed it to Classlaoder, since otherwise we create a new classlaoder on each run, which can really grow the memory.